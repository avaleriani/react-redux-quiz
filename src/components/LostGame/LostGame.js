import React from "react";
import Btn from "../Btn/Btn";
import styles from "./styles.module.css";

/**
 * @param {{restartGame: function, score: number}} props
 * @return HTMLElement
 */
const LostGame = (props) => (
  <div className={styles['lost']}>
    <div className={styles['lost-text']}>
      <span role="img" aria-label="cross-emoji">❌</span>
      <div>Game over </div>
      <span role="img" aria-label="cross-emoji"> ❌</span>
      <br/>
      <div>Your score: {props.score}</div>
    </div>
    <Btn action={props.restartGame} text="Restart"/>
  </div>
);

export default LostGame;