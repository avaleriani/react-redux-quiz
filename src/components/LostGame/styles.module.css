.lost {
    display: flex;
    flex-direction: column;
}

.lost-text {
    border: 2px solid #ff0000;
    padding: 30px;
    text-align: center;
    margin-bottom: 54px;
    font-size: 1.2rem;
}