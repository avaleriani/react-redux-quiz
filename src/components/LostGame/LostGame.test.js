import React from 'react';
import renderer from 'react-test-renderer';
import LostGame from './LostGame';

describe('LostGame component', () => {

  test('renders correctly without props', () => {
    const LostGameComponent = renderer.create(<LostGame/>).toJSON();
    expect(LostGameComponent).toMatchSnapshot();
  });

  test('renders with props', () => {
    const props = {
      score: 22, restartGame: () => {
      }
    };
    const LostGameComponent = renderer.create(<LostGame {...props}/>).toJSON();
    expect(LostGameComponent).toMatchSnapshot();
  });
});