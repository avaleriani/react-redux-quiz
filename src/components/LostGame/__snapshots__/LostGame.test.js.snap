// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LostGame component renders correctly without props 1`] = `
<div
  className="lost"
>
  <div
    className="lost-text"
  >
    <span
      aria-label="cross-emoji"
      role="img"
    >
      ❌
    </span>
    <span>
      Game over 
    </span>
    <span
      aria-label="cross-emoji"
      role="img"
    >
       ❌
    </span>
    <br />
    <div>
      Your score: 
    </div>
  </div>
  <button
    className="button"
    onClick={null}
    type="button"
  >
    Restart
  </button>
</div>
`;

exports[`LostGame component renders with props 1`] = `
<div
  className="lost"
>
  <div
    className="lost-text"
  >
    <span
      aria-label="cross-emoji"
      role="img"
    >
      ❌
    </span>
    <span>
      Game over 
    </span>
    <span
      aria-label="cross-emoji"
      role="img"
    >
       ❌
    </span>
    <br />
    <div>
      Your score: 
      22
    </div>
  </div>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Restart
  </button>
</div>
`;
